### PR TITLE
Story 10

### PR DESCRIPTION
**Pull request by agallant121**
Complete user story 10

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -23,6 +23,7 @@ group :development, :test do
   gem 'pry'
   gem 'simplecov'
   gem 'shoulda-matchers'
+  gem 'bcrypt'
```
Comment (agallant121) [inline Gemfile:26]: Added bcrypt for the password attribute to function

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class UsersController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.create(user_params)
```
Comment (agallant121) [inline app/controllers/users_controller.rb:7]: Brought in user information from new form and then if it was valid, which it was, it saved to the DB and did the proper redirection / showed the flash message on the profile page

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
```
Comment (agallant121) [inline app/models/user.rb:1]: Add user model with proper validations from user model spec

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  validates_presence_of :city
+  validates_presence_of :state
+  validates_presence_of :zip
+  validates_presence_of :email
+  validates_presence_of :password, require: true
+  # attr_accessor :password_confirmation
+
+  validates_uniqueness_of :email
+
+  has_secure_password
```
Comment (agallant121) [inline app/models/user.rb:13]: This was needed to make password validation work

File: config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-
+  get '/', to: 'welcome#index'
```
Comment (agallant121) [inline config/routes.rb:3]: Created just to have a welcome page

File: db/migrate/20200218224518_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
```
Comment (agallant121) [inline db/migrate/20200218224518_create_users.rb:1]: Add user table to the DB

File: spec/features/users/new_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+
+describe 'new user' do
+  	it 'I should be able to register as a new user' do
+
+  		visit '/'
```
Comment (agallant121) [inline spec/features/users/new_spec.rb:6]: Visit welcome index view so we can click on the register new user link

File: spec/features/users/new_spec.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  		fill_in :city, with: city
+  		fill_in :state, with: state
+  		fill_in :zip, with: zip
+  		fill_in :email, with: email
+  		fill_in :password, with: password
+  		fill_in :password_confirmation, with: password_confirmation
+
+  		click_on 'Create User'
+  		expect(current_path).to eq('/profile')
+      expect(page).to have_content("#{name} is now logged in")
```
Comment (agallant121) [inline spec/features/users/new_spec.rb:32]: This is what we expect the flash message to say

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
```
Comment (agallant121) [inline spec/models/user_spec.rb:1]: Add model spec for users to check validations

**Review by meghanstovall — APPROVED** [review]

**Pull request merged**